homeassistant/components/zeroconf/__init__.py

```diff
@@ -541,7 +541,7 @@ def info_from_service(service: AsyncServiceInfo) -> ZeroconfServiceInfo | None:
             if isinstance(value, bytes):
                 properties[key] = value.decode("utf-8")
 
-    if not (addresses := service.addresses):
+    if not (addresses := service.addresses or service.parsed_addresses()):
```

Comment:
We need a test for this case

Reply:
Added IPv6-only test, there was one for "prefers IPv4" already.